### PR TITLE
[ssl] bestrq support

### PR DESCRIPTION
**Pull request by Mddct**
[bestrq paper](https://arxiv.org/pdf/2202.01855.pdf)
[usm paper](https://arxiv.org/abs/2303.01037)

NOTE: only test encoder_type == 'conformer/transformer'
no benchmark for now

### CODE REVIEW

File: wenet/ssl/bestrq/mask.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    # num_masked <= seq_length
+    if num_masked_spans * mask_length > seq_length:
+        num_masked_spans = seq_length // mask_length
+
+    pdf = torch.ones(batch_size, seq_length - (mask_length - 1), device=device)
+    mask_idxs = _sampler(pdf, num_masked_spans, device=device)
+
+    mask_idxs = mask_idxs.unsqueeze(-1).repeat(1, 1, mask_length).view(
+        batch_size, num_masked_spans * mask_length)  # [B,1]
```
Comment (xingchensong) [inline wenet/ssl/bestrq/mask.py:35]: [B,1] ?

Comment (Mddct) [reply wenet/ssl/bestrq/mask.py:36]: sorry my mistakes  [B, num_masked_spans * mask_length ]

**Review by robin1001 — APPROVED** [review]
